addon/computeds/duration.js

```diff
@@ -5,30 +5,25 @@ import isDescriptor from '../utils/is-descriptor';
 
 const { get, computed:emberComputed } = Ember;
 
-function computedDuration(val, maybeUnits) {
-  const numArgs = arguments.length;
-  const args = [val];
-
-  const computed = emberComputed(val, {
+function computedDuration(...args) {
+  const computed = emberComputed({
     get() {
-      const momentArgs = [get(this, val)];
-
-      if (numArgs > 1) {
-        const desc = isDescriptor.call(this, maybeUnits);
-        const input = desc ? get(this, maybeUnits) : maybeUnits;
+      const propertyValues = args.map((arg) => {
+        const desc = isDescriptor.call(this, arg);
+        const argType = typeof get(this, arg);
 
-        if (desc && computed._dependentKeys.indexOf(maybeUnits) === -1) {
-          computed.property(maybeUnits);
+        if (desc && computed._dependentKeys.indexOf(arg) === -1 || argType !== 'undefined') {
+          computed.property(arg);
         }
 
-        momentArgs.push(input);
-      }
+        return desc || argType !== 'undefined' ? get(this, arg) : arg;
```

Comment:
@stefanpenner is there an API equivalent to obj.hasOwnProperty that works with Ember objects?  Right now I'm relying on typeof undefined :/